### DISCUSSION TITLE
RP2040: Example "msc_sdfat" wont compile

### DISCUSSION THREAD

**Discussion by madias123** [adafruit/Adafruit_TinyUSB_Arduino#306] [General]
Hello,
(Using the Pico core: https://github.com/earlephilhower/arduino-pico)
I cannot get the (for v2 sdfat adapted) example of "msc_sdfat" working:
[https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/master/examples/MassStorage/msc_sdfat/msc_sdfat.ino](url)

Line 164: `sd.cacheClear();`
Quits with:
`In file included from C:\Users\matth\AppData\Local\Temp\.arduinoIDE-unsaved202356-5956-1sr0qn4.r76ai\msc_sdfat\msc_sdfat.ino:17:
c:\Users\matth\Documents\Arduino\libraries\SdFat\src/SdFat.h:448:2: warning: #warning File not defined because __has_include(FS.h) [-Wcpp]
  448 | #warning File not defined because __has_include(FS.h)
      |  ^~~~~~~
C:\Users\matth\AppData\Local\Temp\.arduinoIDE-unsaved202356-5956-1sr0qn4.r76ai\msc_sdfat\msc_sdfat.ino: In function 'void msc_flush_cb()':
C:\Users\matth\AppData\Local\Temp\.arduinoIDE-unsaved202356-5956-1sr0qn4.r76ai\msc_sdfat\msc_sdfat.ino:164:6: error: 'SdFat' {aka 'class SdFs'} has no member named 'cacheClear'
  164 |   sd.cacheClear();
      |      ^~~~~~~~~~
exit status 1
Compilation error: 'SdFat' {aka 'class SdFs'} has no member named 'cacheClear'`

So there is no cacheClear() function available.

Things I tried:
Using the newest sdfat library instead of the build in one
Editing the sdfatconfig.h file, like activating
#define USE_BLOCK_DEVICE_INTERFACE 1

So there must be a stupid thing, I forgotten...

Thank you
regards Matthias

**Comment by hathach** [comment]
you need to use the Adafruit Fork of SdFat https://github.com/adafruit/SdFat

**Reply by madias123** [reply]
Ok, thank you, so meanwhile there are - at least - 3 versions for the RP2040 (and ESPxxxx) of the Sdfat by Greiman:
1) Original one
2) build in core (ESP8266SdFat)
3) Adafruit Sdfat Library
